File: packages/sdk-py/theatrical/types/pagination.py
"""Pagination types."""

from __future__ import annotations

from enum import Enum
from typing import Generic, Optional, TypeVar

from theatrical.types.base import ApiModel


T = TypeVar("T")


class PaginationStrategy(str, Enum):
    CURSOR = "cursor"
    OFFSET = "offset"


class PaginationParams(ApiModel):
    limit: Optional[int] = None
    cursor: Optional[str] = None
    offset: Optional[int] = None


class PaginatedResponse(ApiModel, Generic[T]):
    data: list[T]
    total: int
    has_more: bool
    next_cursor: Optional[str] = None
    next_offset: Optional[int] = None
    strategy: PaginationStrategy

File: packages/sdk-csharp/src/Theatrical.Sdk/Types/Pagination.cs
namespace Theatrical.Sdk.Types;

public sealed record PaginatedResponse<T>
{
    public required T[] Data { get; init; }
    public required int Total { get; init; }
    public required bool HasMore { get; init; }
    public string? NextCursor { get; init; }
    public int? NextOffset { get; init; }
    public string? Strategy { get; init; }
}
